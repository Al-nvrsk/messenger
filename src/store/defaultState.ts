export interface AppState {
  appIsInited: boolean
  isLoading: boolean
  isAuth: boolean
  loginFormError: string | null
  user: User | null
  chats: object[] | null
}

export const defaultState: AppState = {
  appIsInited: false,
  isLoading: false,
  isAuth: false,
  loginFormError: null,
  user: null,
  chats: null
}
