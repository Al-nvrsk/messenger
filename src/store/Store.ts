import set from 'utils/helper/set'
import EventBus from 'utils/EventBus'
import { defaultState } from './defaultState'

export enum StoreEvents {
  Updated = 'updated',
  Deleted = 'deleted'
}

class Store extends EventBus {
  private readonly state: Indexed = defaultState

  public getState (): Indexed {
    return this.state
  }

  public setState (path: string, value: unknown): void {
    set(this.state, path, value)
    this.emit(StoreEvents.Updated)
  };
}

export default new Store()
