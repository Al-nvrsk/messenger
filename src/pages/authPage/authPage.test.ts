import '@testing-library/jest-dom'
import { getByTestId } from '@testing-library/dom'
import AuthPage from './authPage'
import mockedRenderDOM from 'test/mockedRenderDOM'
import store from 'store/Store'
import mockedRegisteredComp from 'test/mockedRegisteredComp'

jest.mock('utils/renderDOM')

beforeEach(() => {
  store.setState('isAuth', true)
  mockedRegisteredComp()
  mockedRenderDOM(new AuthPage(store))
})

describe('pages/Authpage', () => {
  it('should render authPage', () => {
    const authPageBody = document.querySelector('.authwindow')
    expect(authPageBody).toMatchSnapshot()
  })
  it('should render authPageButton element', () => {
    expect(getByTestId(document.body, 'authPageButton')).toBeInTheDocument()
  })
})
