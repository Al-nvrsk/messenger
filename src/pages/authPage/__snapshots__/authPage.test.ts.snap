// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Authpage should render authPage 1`] = `
<div
  class="authwindow"
>
  
        
  <h1>
     Sign in 
  </h1>
  

  <form
    class="authform"
  >
    
        
    <div>
                
      <div>
        
      
        <div
          class="controlledInput"
        >
          
      
          <div
            class="controlledInputLabel"
          >
            
        Input your login
      
          </div>
          
      
          <div
            class="controlledInputField"
          >
            
      
            <input
              class="inputBase"
              name="login"
              placeholder="login"
              ref=""
              type="text"
              value=""
            />
            
      
          </div>
          
        
          <hr
            class="controlledInputLine"
          />
          
      
          <div
            class="errorrs"
          >
            
      
            <div
              class="errorBase"
            />
            
      
          </div>
          
      
        </div>
        
      
      </div>
       
      <div>
        
      
        <div
          class="controlledInput"
        >
          
      
          <div
            class="controlledInputLabel"
          >
            
        Input your password
      
          </div>
          
      
          <div
            class="controlledInputField"
          >
            
      
            <input
              class="inputBase"
              name="password"
              placeholder="password"
              ref=""
              type="password"
              value=""
            />
            
      
          </div>
          
        
          <hr
            class="controlledInputLine"
          />
          
      
          <div
            class="errorrs"
          >
            
      
            <div
              class="errorBase"
            />
            
      
          </div>
          
      
        </div>
        
      
      </div>
      

          
      <div
        class="authPageButton"
        data-testid="authPageButton"
      >
        
            
        <button
          class="buttonAccept"
          type="submit"
        >
           Enter 
        </button>
        
            
        <button
          class="buttonAccept"
          type="button"
        >
           Create account 
        </button>
        

      </div>
                
    </div>
    
    
  </form>
  
        
</div>
`;
